projects/ui/src/Utility/AdminUtility/RateLimitEditor.tsx

```diff
@@ -190,14 +187,19 @@ export const RateLimitEditor = ({
               <Text size="md">
                 <label htmlFor="unit-input">Unit</label>
               </Text>
-              <Input
+              <Select
                 required
                 disabled={!isEditingRateLimit}
                 id="unit-input"
+                data={rateLimitUnitOptions}
+                onChange={(value: RateLimitUnit | null) => {
+                  if (!!value) {
+                    setUnit(value);
+                  }
+                }}
```

Comment:
This was mostly a readability change. In the UI codebase (both here and GME), we try to be explicit when checking variable truthiness by converting it to a boolean first, like this (`!!value` is the same as `Boolean(value)`).